### PR TITLE
Fix dependent key value with MySQL client

### PR DESCRIPTION
**Pull request by ricardogama**
This PR fixes the format of the dependent key building the delete queries, adding tests for the **MySQL** client.

### CODE REVIEW

**Comment by renarsvilnis** [comment]
Run tests against MySQL 5.5.42 locally, everything passed.

Just noticed that MySQL test file had PostgreSQL comment and description left from copy-pasting.

``` javascript
// test/mysql/index.js

/**
 * Test PostgreSQL client.
 */

describe('with PostgreSQL client', () => {
```

**Comment by ricardogama** [comment]
@renarsvilnis Thanks! I was repeating a lot of code in both files so I decided to test both clients on _test/index.js_ and reuse the repeated code with the `initialize` function, take a look.

**Comment by renarsvilnis** [comment]
All good!

**Comment by renarsvilnis** [comment]
When are you planing to merge the PR and bump the version?

**Comment by ricardogama** [comment]
I would like to have @Bartinger input regarding `relatedData.key('foreignKey')`. If he does not reply until the end of the day I'll make a release and solve that issue in another PR.

File: src/index.js [hunk 1]
```diff
@@ -41,32 +42,33 @@ export default Bookshelf => {
...
-    recursiveDeletes(parent, options) {
+    recursiveDeletes(parent) {
       // Stringify in case of parent being an instance of query.
       const parentValue = typeof parent === 'number' || typeof parent === 'string' ? `'${parent}'` : parent.toString();
 
       // Build delete queries for each dependent.
       const queries = reduce(this.dependencyMap(), (result, dependent) => {
         const tableName = dependent.model.prototype.tableName;
-        const whereClause = `"${dependent.key}" IN (${parentValue})`;
+        const dependentKey = client === 'postgres' ? `"${dependent.key}"` : dependent.key;
```
Comment (nunorafaelrocha) [inline src/index.js:63]: Sort.

File: test/index.js [hunk 2]
```diff
@@ -3,165 +3,335 @@
...
+      await clearTables(repository);
+    });
 
-  it('should not delete models which are not dependent', async () => {
-    const author1 = await Author.forge().save();
-    const author2 = await Author.forge().save();
-    const post1 = await Post.forge().save({ authorId: author1.get('id') });
-    const post2 = await Post.forge().save({ authorId: author2.get('id') });
+    after(async () => {
+     await dropTables(repository);
```
Comment (nunorafaelrocha) [inline test/index.js]: Fix indentation.

File: test/utils/index.js [hunk 3]
```diff
@@ -47,10 +41,43 @@ export async function clearTables() {
...
     .dropTable('Author');
 }
+
+/**
+ * Export `fixtures`.
+ */
+
+export function fixtures(repository) {
+  const Account = repository.Model.extend({ tableName: 'Account' });
+
```
Comment (nunorafaelrocha) [inline test/utils/index.js]: Remove new line.

**Comment by ricardogama** [comment]
@renarsvilnis @Bartinger Released [1.0.1](https://github.com/seegno/bookshelf-cascade-delete/releases/tag/1.0.1) with this fix.

**Comment by renarsvilnis** [comment]
@ricardogama Thanks, works as a charm.

Just wondering, if I want to to stuff on model destroy, e.g. delete file thats connected to model, I should do it on `deleting` event rather then overriding `destroy` method of a model?

**UPDATE:**
Created an issue #10.

**Pull request merged**